models/schemas/achievementSchema.ts

```diff
@@ -0,0 +1,21 @@
+import mongoose, { Mongoose } from 'mongoose';
+import AchievementType from '../../constants/achievementType';
+import Achievement from '../db/achievement';
+
+
+const AchievementSchema = new mongoose.Schema<Achievement>({
```

Comment:
is this schema comprehensive enough? I would imagine that it would have some generic data property that would have the properties for the different types of achievements (frontend could switch on the type)